chains/config.go

```diff
@@ -0,0 +1,108 @@
+package chains
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	Chains []RawChainConfig `mapstructure:"chains"`
+}
+
+type RawChainConfig struct {
+	GeneralChainConfig `mapstructure:",squash"`
+	Opts               map[string]string
+}
+
+type GeneralChainConfig struct {
+	Name     string `mapstructure:"name"`
+	Type     string `mapstructure:"type"`
+	Id       uint8  `mapstructure:"id"`
+	Endpoint string `mapstructure:"endpoint"`
+	From     string `mapstructure:"from"`
+}
+
+func NewConfig() *Config {
+	return &Config{
+		Chains: []RawChainConfig{},
+	}
+}
+
+func GetConfig(path string, name string) (*Config, error) {
+	var config Config
+
+	viper.AddConfigPath(path)
+	viper.SetConfigName(name)
+	viper.SetConfigType("json")
+
+	err := viper.ReadInConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read in the config file, error: %w", err)
+	}
```

Comment:
Completely optional:

Calls like this that only return an error can be rewritten

```go
if err := viper.ReadInConfig(); err != nil {
// handle err
}
```